tsdb/head.go

```diff
@@ -1609,6 +1619,22 @@ func (h *Head) getOrCreateWithID(id chunks.HeadSeriesRef, hash uint64, lset labe
 	return s, true, nil
 }
 
+func (h *Head) mmapHeadChunks() {
+	var count int
+	for i := 0; i < h.series.size; i++ {
+		h.series.locks[i].RLock()
+		for _, all := range h.series.hashes[i] {
+			for _, series := range all {
+				series.Lock()
+				count = series.mmapHeadChunks(h.chunkDiskMapper)
```

Comment:
I suggest gathering all the counts and do a single metric update of `mmapChunksTotal` at the end of the function. We want to get rid of holding the `h.series.locks` locks as fast as possible.
```suggestion
				count += series.mmapHeadChunks(h.chunkDiskMapper)
```